dashboard/src/reducers/namespace.test.ts

```diff
@@ -66,3 +66,31 @@ describe("namespaceReducer", () => {
     });
   });
 });
+
+context("when SET_DEFAULT_NAMESPACE", () => {
+  it("set current namespace if it's empty", () => {
```

Comment:
It's a bit confusing to read that `SET_DEFAULT_NAMESPACE` is actually setting the current namespace - I think this confusion is there because we're mixing concerns. I think what I would do personally is:

1) Have a `NAMESPACE_RECEIVED` action defined in `actions/auth.ts` (with it's corresponding `namespaceReceived()`) as the auth module the subject/cause of receiving the data, and dispatch that where you're currently dispatching the imported `SET_DEFAULT_NAMESPACE/setDefaultNamespace`.
2) Update any reducers which are interested in this - which in this case, is the namespace reducer. The name of the action does not imply what reducers will do with the info, nor restrict who it is for, each module can choose to do what they want (which in this case, will be the namespace reducer updating its own current namespace state only if it's not currently set).

This is one of the reasons for redux actions - separating concerns... all the auth module cares about here is publishing the fact that it received the info (not that it needs to be set as a default or current in a separate module).

The tests here would not change, other than the action type, just the implementation is slightly different.

See what you think.

Reply:
what I wanted to distinguish is between a manual change of the namespace (changing it in the namespace selector) vs the automatic namespace from the token but I agree with your reasoning.